core/client_config.go

```diff
@@ -40,22 +40,24 @@ type ClientConfig struct {
 	Timeout   time.Duration `koanf:"timeout"`
 }
 
-// DefaultClientConfig creates a new CLI client config with default values set.
-func DefaultClientConfig() ClientConfig {
-	return ClientConfig{
-		Address:   defaultAddress,
-		Verbosity: defaultLogLevel,
-		Timeout:   defaultClientTimeout,
+// NewClientConfigForCommand loads the config map into a command
+func NewClientConfigForCommand(cmd *cobra.Command) (ClientConfig, error) {
```

Comment:
What about a panic/fatal instead of the errors? Since the caller will always be a client command it can just exit right away when the config is incorrect. Saves a lot of test code as well.

Reply:
I don't like the stacktraces which appear after a panic. It feels like not handling an error and just returning a 500 because is saves a lot of code.